test/cpp/api/modules.cpp

```diff
@@ -3200,6 +3201,9 @@ namespace detail {
       } else {
         std::uniform_int_distribution<int> d(5, 20);
         kv_dim = d(generator);
+        while (kv_dim == d_model) {
```

Comment:
what's this for?

Reply:
This branch was aimed to test kv_dim != d_model case, but the generator d could generate a same value as d_model, for eg. 9, so I add this while.